### ISSUE TITLE
[19.2] Object null is not iterable (cannot read property Symbol(Symbol.iterator))

### ISSUE THREAD

**Issue by EelcoLos** [nrwl/nx#26403]
### Current Behavior

running any `nx` command after migrating to 19.2 results in the error
>  NX   object null is not iterable (cannot read property Symbol(Symbol.iterator))

### Expected Behavior

`nx` commands run properly

### Steps to Reproduce

1. from `19.1.1`, migrate to `19.2.0` with `nx migrate latest; npm i; nx migrate --run-migrations --create-commits`
1. Run a nx command, ie: `nx run-many -t lint --fix`

### Nx Report

NOTE: i had to downgrade to 19.1.1 to create this report, then i updated the numbers to 19.2 of the upgraded nx packages
```shell
NX   Report complete - copy this into the issue template

Node   : 20.14.0
OS     : win32-x64
npm    : 10.2.5

nx (global)        : 19.2.0
nx                 : 19.2.0
@nx/js             : 19.2.0
@nx/jest           : 19.2.0
@nx/linter         : 19.2.0
@nx/eslint         : 19.2.0
@nx/workspace      : 19.2.0
@nx/angular        : 19.2.0
@nx/devkit         : 19.2.0
@nx/eslint-plugin  : 19.2.0
@nx/playwright     : 19.2.0
@nx/plugin         : 19.2.0
@nrwl/tao          : 19.2.0
@nx/web            : 19.2.0
@nx/webpack        : 19.2.0
typescript         : 5.4.5
---------------------------------------
Registered Plugins:
@nx/eslint/plugin
@nx-dotnet/core
@nx/jest/plugin
---------------------------------------
Community plugins:
@fortawesome/angular-fontawesome : 0.15.0
@ngrx/component                  : 18.0.0-beta.1
@ngrx/component-store            : 18.0.0-beta.1
@ngrx/effects                    : 18.0.0-beta.1
@ngrx/entity                     : 18.0.0-beta.1
@ngrx/router-store               : 18.0.0-beta.1
@ngrx/schematics                 : 18.0.0-beta.1
@ngrx/store                      : 18.0.0-beta.1
@ngrx/store-devtools             : 18.0.0-beta.1
@nx-dotnet/core                  : 2.2.0
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

### Additional Information

As was pointed out on the migration, I also added `.nx/workspace-data` to `.gitignore` too

**Comment by IgorVyatkin** [comment]
`nx run-many --targets lint` also has stopped working for me after upgrade to 19.2.0.

I'm getting the following error for each app/lib:
```
×  nx run @company/app-server:lint
    Linting "@mes/app-lims"...

    NX   ENOENT: no such file or directory, stat 'C:\Work\src\C:\Work\src\company.app.server.client\apps\server\eslint.config.js'
```

**Comment by AgentEnder** [comment]
Can you run with --verbose and `NX_VERBOSE_LOGGING=true` to capture a stack trace?

**Comment by EelcoLos** [comment]
> Can you run with --verbose and `NX_VERBOSE_LOGGING=true` to capture a stack trace?

the following is mentioned with `--verbose` and `NX_VERBOSE_LOGGING=true`:
```
{
  stack: 'TypeError: object null is not iterable (cannot read property Symbol(Symbol.iterator))\n' +
    '    at D:\\DigitaalMeten\\node_modules\\nx\\src\\project-graph\\utils\\project-configuration-utils.js:248:53\n' +
    '    at Array.map (<anonymous>)\n' +
    '    at D:\\DigitaalMeten\\node_modules\\nx\\src\\project-graph\\utils\\project-configuration-utils.js:248:49\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
    '    at async Promise.all (index 1)\n' +
    '    at async processFilesAndCreateAndSerializeProjectGraph (D:\\DigitaalMeten\\node_modules\\nx\\src\\daemon\\server\\project-graph-incremental-recomputation.js:148:43)\n' +
    '    at async getCachedSerializedProjectGraphPromise (D:\\DigitaalMeten\\node_modules\\nx\\src\\daemon\\server\\project-graph-incremental-recomputation.js:46:16)\n' +
    '    at async handleRequestProjectGraph (D:\\DigitaalMeten\\node_modules\\nx\\src\\daemon\\server\\handle-request-project-graph.js:12:24)\n' +
    '    at async handleResult (D:\\DigitaalMeten\\node_modules\\nx\\src\\daemon\\server\\server.js:130:16)\n' +
    '    at async handleMessage (D:\\DigitaalMeten\\node_modules\\nx\\src\\daemon\\server\\server.js:89:9)',
  message: 'object null is not iterable (cannot read property Symbol(Symbol.iterator))'
}
```

**Comment by yjaaidi** [comment]
Just stumbled upon the same error.

This happens because options are hashed before being normalized so the hash function fails when options are `undefined`.
Cf. https://github.com/nrwl/nx/blob/7495f0664b19e8fa32ef693f43d709173b6a2bc4/packages/eslint/src/plugins/plugin.ts#L162

The temporary workaround is to pass an empty options object in `nx.json`.
```diff
- "plugins": ["@nx/eslint/plugin"]
+ "plugins": [{"plugin": "@nx/eslint/plugin", "options": {}}]
```

**Comment by EelcoLos** [comment]
> Just stumbled upon the same error.
> 
> This happens because options are hashed before being normalized so the hash function fails when options are `undefined`. Cf.
> 
> https://github.com/nrwl/nx/blob/7495f0664b19e8fa32ef693f43d709173b6a2bc4/packages/eslint/src/plugins/plugin.ts#L162
> 
> The temporary workaround is to pass an empty options object in `nx.json`.
> 
> ```diff
> - "plugins": ["@nx/eslint/plugin"]
> + "plugins": [{"plugin": "@nx/eslint/plugin", "options": {}}]
> ```

can confirm, after adding that and opening a new shell, this works again

**Comment by cyrus-za** [comment]
This solves it for me too. Thanks

**Comment by AgentEnder** [comment]
There are a few other plugins with the issue, we'll get it fixed.

**Comment by AgentEnder** [comment]
This is fixed by a combination of #26455 and https://github.com/nrwl/nx/pull/26456
